osl/interface_linux.go

```diff
@@ -335,6 +350,18 @@ func setInterfaceIP(iface netlink.Link, i *nwIface) error {
 	return netlink.AddrAdd(iface, ipAddr)
 }
 
+func setInterfaceIPAliases(iface netlink.Link, i *nwIface) error {
+	if i.IPAliases() == nil {
+		return nil
+	}
+
+	for _, ip := range i.IPAliases() {
+		ipAddr := &netlink.Addr{IPNet: ip, Label: ""}
+		netlink.AddrAdd(iface, ipAddr)
```

Comment:
If any of the `AddrAdd()` fails, we need to stop here and return error.
Either all all addresses were successfully added, or it's a failure.